keps/sig-scheduling/5027-dra-admin-controlled-device-attributes/README.md

```diff
@@ -350,7 +354,32 @@ type DevicePatchFilter struct {
     // we might have one entry which uses CEL while another uses a different, yet to be
     // defined mechanism.
 }
+```
+
+To distinguish intentionally empty attributes from attributes which
+have some future, unknown content and thus only seem to be empty to an older
+client, the DeviceAttribute struct get extended:
+
+```golang
+// DeviceAttribute must have exactly one field set.
+type DeviceAttribute struct {
+    // The Go field names below have a Value suffix to avoid a conflict between the
+	// field "String" and the corresponding method. That method is required.
+	// The Kubernetes API is defined without that suffix to keep it more natural.
+
+    ...
+
+    // NullValue, if set, marks an intentionally empty attribute.
+    //
+    // May be used inside a ResourceSlicePatch to remove attributes,
+    // but not in a ResourceSlice.
+    //
+    // +optional
+    // +oneOf=ValueType
+    NullValue *NullValue `json:"null,omitempty"`
```

Comment:
In the CEL environment, will null values be omitted entirely from the `device.attributes` map if we don't handle them specially? If not, enabling that might be worthwhile since I can see that being a little more ergonomic for authors of CEL expressions than if they have to handle "explicit null" and "actually undefined" differently.

Reply:
I guess my question here boils down to "how does a Go map with a value of `nil` manifest in the CEL environment, and is that different from a Go map without that key/value pair at all?"

Reply:
Because ResourceSlices contain no null values and null values in a ResourceSlicePatch cause the attribute to be removed, there's never a situation where a CEL expression gets a null value when looking up an attribute. That is deliberate: we already have "attribute not set in map", we don't need "attribute set with null value". That is semantically so close that I don't see the need.